### ISSUE TITLE
Robustness to Pruning Predicts Generalization in Deep Neural Networks

### ISSUE THREAD

**Issue by Hiroki11x** [Hiroki11x/Papers#74]
https://openreview.net/forum?id=1P2KAvsE59b

Keywords: Generalization, Pruning, Generalization Measures
Abstract: Why over-parameterized neural networks generalize as well as they do is a central concern of theoretical analysis in machine learning today. Following Occam's razor, it has long been suggested that simpler networks generalize better than more complex ones. Successfully quantifying this principle has proved difficult given that many measures of simplicity, such as parameter norms, grow with the size of the network and thus fail to capture the observation that larger networks tend to generalize better in practice.
In this paper, we introduce a new, theoretically motivated measure of a network's simplicity: the smallest fraction of the network's parameters that can be kept while pruning without adversely affecting its training loss. We show that this measure is highly predictive of a model's generalization performance across a large set of convolutional networks trained on CIFAR-10. Lastly, we study the mutual information between the predictions of our new measure and strong existing measures based on models' margin, flatness of minima and optimization speed. We show that our new measure is similar to -- but more predictive than -- existing flatness-based measures.

```
オーバーパラメータ化されたニューラルネットワークがなぜこれほどまでに一般化するのかは、今日の機械学習における理論分析の中心的な関心事である。オッカムのカミソリに従い、単純なネットワークの方が複雑なネットワークよりも一般化しやすいと言われてきた。しかし、パラメータノルムのような単純性を表す多くの尺度は、ネットワークの大きさに応じて大きくなるため、実際には大きなネットワークの方が一般化しやすいという観察結果を捉えることができないため、この原理をうまく定量化することは困難でした。
本論文では、理論的に動機づけられた新しいネットワークの単純性の尺度を紹介する。それは、学習損失に悪影響を与えることなく、刈り込み中に維持できるネットワークのパラメータの最小の割合である。この指標は、CIFAR-10で学習された大規模な畳み込みネットワークのセットにおいて、モデルの汎化性能を高度に予測することが示された。最後に，我々の新しい尺度と，モデルのマージン，最小値の平坦性，最適化速度に基づく既存の強力な尺度との間の相互情報を研究する．その結果、我々の新しい尺度は、既存の平坦性に基づく尺度と似ているが、それよりも予測性が高いことを示した。
```
